src/thingset.h

```diff
@@ -323,17 +338,390 @@ typedef struct TsDataNode {
 
 } TsDataNode;
 
+/**
+ * Main ThingSet data structure
+ *
+ * Stores and handles all data exposed to different communication interfaces
+ */
+typedef struct ts_object {
+    /**
+     * Array of nodes database provided during initialization
+     */
+    TsDataNode *data_nodes;
+
+    /**
+     * Number of nodes in the data_nodes array
+     */
+    size_t num_nodes;
+
+    /**
+     * Pointer to request buffer (provided in process function)
+     */
+    uint8_t *req;
+
+    /**
+     * Length of the request
+     */
+    size_t req_len;
+
+    /**
+     * Pointer to response buffer (provided in process function)
+     */
+    uint8_t *resp;
+
+    /**
+     * Size of response buffer (i.e. maximum length)
+     */
+    size_t resp_size;
+
+    /**
+     * Pointer to the start of JSON payload in the request
+     */
+    char *json_str;
+
+    /**
+     * JSON tokes in json_str parsed by JSMN
+     */
+    jsmntok_t tokens[TS_NUM_JSON_TOKENS];
+
+    /**
+     * Number of JSON tokens parsed by JSMN
+     */
+    int tok_count;
+
+    /**
+     * Stores current authentication status (authentication as "normal" user as default)
+     */
+    uint16_t _auth_flags;
+} ts_object_t;
```

Comment:
In order to be constistent with naming of typedef-ed structs (see also [here](https://github.com/LibreSolar/coding-style)) I suggest following naming:
```C
typedef struct thingset_ctx {
    // ...
} ThingSetCtx;
```
For native Zephyr one would probably use `struct thingset_ctx` then instead of the typedef.